typescript/src/server/rxjs-socket.ts

```diff
@@ -1,34 +1,39 @@
-import { Subject } from "rxjs";
-import WebSocket from "ws";
-import { createMessage, Message } from "../message";
+import { Observable } from "rxjs";
+import WebSocket, { RawData } from "ws";
+import { Message } from "../message";
 import { BaseSocket, RxSocket } from "./universal-types";
 
-export default class SocketImpl implements RxSocket, BaseSocket {
-    events: Subject<Message>;
+export function fromWebSocket(socket: WebSocket): Observable<Message> {
```

Comment:
Wrapping of WebSocket as an observable moved out to its own function (I would test this if I was unit testing)